pkg/controller/migmigration/stage.go

```diff
@@ -91,6 +90,46 @@ func BuildStagePods(labels map[string]string,
 	return stagePods
 }
 
+// get the running application pods with attached PVC
+func GetApplicationPodsWithStageLabels(labels map[string]string, pvcMapping map[k8sclient.ObjectKey]migapi.PV, list *[]corev1.Pod) []corev1.Pod {
+	applicationPodsWithStageLabel := []corev1.Pod{}
+	for _, pod := range *list {
+		foundVolume := true
+		excludePVC := ""
+		for _, volume := range pod.Spec.Volumes {
+			if volume.PersistentVolumeClaim == nil {
+				continue
+			}
+			claimKey := k8sclient.ObjectKey{
+				Name:      volume.PersistentVolumeClaim.ClaimName,
+				Namespace: pod.Namespace,
+			}
+			pv, found := pvcMapping[claimKey]
+			if !found {
+				continue
```

Comment:
You probably want to add anything from here to excludePVC  too. If something gets to this line, then it's a PVC (so the type check won't catch it), which means the PVC isn't being migrated but the pod still has it.